csrc/root_domain_map.cpp

```diff
@@ -229,6 +229,30 @@ std::unordered_map<IterDomain*, IterDomain*> PairwiseRootDomainMap::map(
     return dom_map;
   }
 
+  if (SdpaFwdOp* op = dynamic_cast<SdpaFwdOp*>(consumer_tv_->definition())) {
+    // Consumers: output = [N, H, L, Ev], logsumexp = [N, H, L], cum_seq_q/k =
+    // [N + 1,] Producers: query = [N, H, L, E], key = [N, H, S, E], value = [N,
+    // H, S, Ev]
+
+    // Map N, H from any input (query/key/value)
+    for (auto idx : c10::irange(consumer_root.size())) {
+      if (idx < 2) {
```

Comment:
For `idx == 0` and `consumer_tv_->sameAs(op->output(2)) || consumer_tv_->sameAs(op->output(3))`, these should not map since the extents differ by 1. I would just put a check before this that consumer is the output or logsumexp. Btw you might want to add accessors like `op->logSumExp()` to make this new condition more readable.

Reply:
Even if I use `resize`? 
I saw an error that there was no mapped iterdomain from producer.

Reply:
Oh ok so you are mapping the consumer root then you have an rfactor domain for that consumer which uses a Resize?

Reply:
That is right then.